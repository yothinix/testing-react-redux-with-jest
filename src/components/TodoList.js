import React from 'react'

import TodoItem from './TodoItem'

const TodoList = props => (
  <ul className="todo-list">
    (props.todos.map(todo =>
      <TodoItem key={todo.id} {...todo} />
    )}
  </ul>
)

export default TodoList
