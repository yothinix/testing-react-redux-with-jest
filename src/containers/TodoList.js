import { connect } from 'react-redux'
import TodoList from '../components/TodoList'
import {
  selectTodos,
} from '../selectors/todosSelector'

export default connect(selectTodos)(TodoList)

