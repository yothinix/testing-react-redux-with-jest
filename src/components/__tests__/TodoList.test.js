import React from 'react'
import { shallow } from 'enzyme'
import TodoList from '../TodoList'

describe('TodoList', () => {
  let component
  let props

  beforeEach(() => {
    props = {
      todos: [
      { id: 1, text: 'Feed cat', complete: false },
      { id: 2, text: 'Drink coffee', complete: true },
      ],
    }
  })

  it('matches its snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('renders correct structure', () => {
    expect(component.is('ul')).toBe(true)
    expect(component.hasClass('todo-list')).toBe(true)
  })

  it('renders TodoItem from props', () => {
    expect(component.find('TodoItem').length)
      .toBe(props.todos.length)
  })
})
