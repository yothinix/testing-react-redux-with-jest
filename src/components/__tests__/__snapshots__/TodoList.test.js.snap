// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList matches its snapshot 1`] = `
<ul
  className="todo-list"
>
  <TodoItem
    complete={false}
    id={1}
    text="Feed cat"
  />
  <TodoItem
    complete={true}
    id={2}
    text="Drink coffee"
  />
</ul>
`;
